SPHINXsys/src/shared/particle_dynamics/solid_dynamics/contact_dynamics.cpp

```diff
@@ -21,7 +21,13 @@ namespace SPH
 			: LocalDynamics(self_contact_relation.sph_body_), SolidDataInner(self_contact_relation),
 			  mass_(particles_->mass_)
 		{
-			particles_->registerVariable(self_contact_density_, "SelfContactDensity");
+            if(auto ptr = particles_->getVariableByName<Real>("SelfContactDensity"))
```

Comment:
Here the variable registration does not follow the registration rules.

Reply:
That's to allow multiple contact relation, as a workaround to problem #147 which invalidly prevents multiple contact topology over time.

Reply:
@Xiangyu-Hu what's the issue here specifically? This is really needed usage-wise as time-dependent contact changes are required for use cases such as stent compression using the virtual catheter method.